contrib/!source-control/github/packages.el

```diff
@@ -86,3 +88,46 @@
         "gfC" 'spacemacs/git-link-commit-copy-url-only)
       ;; default is to open the generated link
       (setq git-link-open-in-browser t))))
+
+;; magit-gh-pulls has to be loaded via a pre-init hook because the source code
+;; makes assumptions about the status of the magit-mode keymaps that are
+;; incompatible with the spacemacs' evilification feature.
+;; To avoid errors, magit-gh-pulls must be loaded after magit, but before magit
+;; is configured by spacemacs. 
+
+(defun github/init-magit-gh-pulls ())
+
+(defun github/pre-init-magit ()
+  (spacemacs|use-package-add-hook magit
+    :pre-config
+    (progn
+      (use-package magit-gh-pulls
+        ;; only load this config for magit-gh-pulls if both 1 and 2 are true:
+        ;; 1) The package is used - this allows users to :exclude the package
+        ;; normally.
+        ;; 2) Magit-gh-pulls is owned by the github layer - this prevents this
+        ;; configuration from being run if another layer has claimed ownership
+        ;; of magit-gh-pulls, and maintains the spacemacs contract of a single
+        ;; owner per package.
+        :if (and (configuration-layer/package-usedp 'magit-gh-pulls)
+                 (oref (-find (lambda (pkg) (equal (oref pkg :name)
+                                                   'magit-gh-pulls))
+                              configuration-layer--packages)
+                       :owner))
```

Comment:
@syl20bnr I've added an additional check that I believe work in preventing the multiple owner problem.

Earlier you suggested moving the use-package-add-hook into init-magit-gh-pulls; wouldn't that change rely on lazy loading of magit?

I'm happy to make whatever changes you like - at this point I'm honestly more interested in exploration the package management system that spacemacs uses.